### DISCUSSION TITLE
nfdump-1.7.0.1: fatal error: stdatomic.h: No such file or directory

### DISCUSSION THREAD

**Discussion by alan-boey** [phaag/nfdump#363] [nfdump-1.7]
Compiled using "./configure --enable-shared=no --enable-sflow" and getting error during make process.
It does not happen in nfdump-1.6.24. Any help appreciated.

libtool: compile:  gcc -std=gnu11 -DHAVE_CONFIG_H -I. -I../.. -I.. -I../include -I../inline -ggdb -g -O3 -std=gnu11 -Wall -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Wmissing-noreturn -fno-strict-aliasing -pthread -MT nffile.lo -MD -MP -MF .deps/nffile.Tpo -c nffile.c -o nffile.o
nffile.c:38:23: fatal error: stdatomic.h: No such file or directory
compilation terminated.
make[3]: *** [nffile.lo] Error 1
make[3]: Leaving directory `/usr/local/src/nfdump-1.7.0.1/src/lib'
make[2]: *** [all] Error 2
make[2]: Leaving directory `/usr/local/src/nfdump-1.7.0.1/src/lib'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/usr/local/src/nfdump-1.7.0.1'
make: *** [all] Error 2

OS: Linux 3.2.0-5-686-pae #1 SMP Debian 3.2.96-3 i686 GNU/Linux

**Comment by phaag** [comment]
You need to move to a newer compiler. 3.2 is a bit too old.

**Reply by alan-boey** [reply]
Thanks for your response.

I'm using 4.7.2 and it still does not compile. May I know which version would be competable.

gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/i486-linux-gnu/4.7/lto-wrapper
Target: i486-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Debian 4.7.2-5' --with-bugurl=file:///usr/share/doc/gcc-4.7/README.Bugs --enable-languages=c,c++,go,fortran,objc,obj-c++ --prefix=/usr --program-suffix=-4.7 --enable-shared --enable-linker-build-id --with-system-zlib --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --with-gxx-include-dir=/usr/include/c++/4.7 --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --enable-gnu-unique-object --enable-plugin --enable-objc-gc --enable-targets=all --with-arch-32=i586 --with-tune=generic --enable-checking=release --build=i486-linux-gnu --host=i486-linux-gnu --target=i486-linux-gnu
Thread model: posix
gcc version 4.7.2 (Debian 4.7.2-5)

**Reply by gabrielmocan** [reply]
Have you tried to install libatomic1? 

apt install libatomic1

**Reply by alan-boey** [reply]
Hi Gabriel,
Yes I've tried it. Unfortunately I'm not able to find a copy that of that works with Debian Wheezy. Guess I'm stuck with this unless I upgrade my OS which will break some of the legacy intranet applications.

Thanks for your suggestion and help.

**Comment by phaag** [comment]
If you have restrictions with other apps, and can not uograde, use  nfdump-1.6.25. This does not need the stdatomic.h

**Reply by alan-boey** [reply]
Thanks, appreciate it.

**Reply by gabrielmocan** [reply]
You might try to compile the lib as a last stand.
